target/linux/ramips/dts/mt7620n_hootoo_tm05.dts

```diff
@@ -0,0 +1,129 @@
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hootoo,tm05", "ralink,mt7620n-soc";
+	model = "HooToo TM05";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		green {
+			label = "tm05:green";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
```

Comment:
Why? Maybe you want to set up aliases for the boot LED etc.

Reply:
I can if you want, just let me know - no issue here. I went with labels matching the color ... as it's a single (dual-color) LED, wanted to make it clear which color is being enabled by each GPIO.

Reply:
Remove default-on, but add phy0tpt or phy0radio trigger here (first should show on/off and activity, second only on/off).

Reply:
OK, changing default state to trigger. Will let you know ... :-). Thanks!

Reply:
phy0radio works, solid green. But phy0tpt not working LED never on - is that the correct name? Thanks!

Reply:
OK, manually changed this in the web interface ... as below. Any idea how to set this in the code? Thanks!
![image](https://user-images.githubusercontent.com/8756384/71518210-63df0180-2877-11ea-97fe-bc93aff686b9.png)

Reply:
Hmmm ... OK, checked uci (uci show system | grep "system.led"),
```
system.led_wifi_led=led
system.led_wifi_led.name='wifi'
system.led_wifi_led.default='0'
system.led_wifi_led.sysfs='tm05:green:wifi'
system.led_wifi_led.trigger='netdev'
system.led_wifi_led.mode='tx' 'rx'
system.led_wifi_led.dev='wlan0'
```
I can set the trigger to netdev in the DTX, but how to add the mode and dev(ice)? Will keep digging.

Reply:
That's a leftover from the 01_leds setup. When you use the DT trigger, you won't have an entry in /etc/config/system at all. Make sure you use "sysupgrade -n" when you test new images, so no old config is preserved.

Reply:
You're right, my mistake - thanks!

I do see the needed info in /sys/devices/platform/leds/leds/tm05:green:wifi, and I can manually make changes (echo, cat to check). Even works with netdev (to put this in place, control tx and rx triggers). Last struggle is the correct "place" / syntax to set the initial values for netdev in the DTS (tx, rx, link, device_name). All working manually, just struggling to find the exact "level" in the DTS, For example, 
https://lore.kernel.org/patchwork/patch/1050531/

Thanks!

Reply:
BTW, in /sys/devices/platform/leds/leds/tm05:green:wifi/trigger, shows the options (and tpt isn't one of them),
`none switch0 timer default-on [netdev] phy0rx phy0tx phy0assoc phy0radio
`

Currently netdev selected (and working) ... again, just need to figure out how to initialize, then it works.

Reply:
So, please use phy0radio.

Reply:
Sure, will do. It seems like the noted patch isn't in OpenWrt ... do I have that right? netdev seems to work fine, just not the init portion of it. Perhaps something to add (separately)? Thanks!

Reply:
FYI, to change it to traffic based afterwards (init script if desired?),
```
cd /sys/devices/platform/leds/leds/tm05:green:wifi
echo netdev > trigger
echo wlan0 > device_name
echo 1 > link
echo 1 > tx
echo 1 > rx
```

Reply:
I think the conclusion here was that phy0radio didn't work, and in the end I was supposed to use "phy0tpt", right?

Reply:
Does it actually have phy0tpt trigger? If not you should use phy0radio.

Reply:
Agreed! When I tried phy0radio, no LED status / blinking (w/traffic). But it does based on phy0tpt. OK?

Reply:
Sorry, it's been a while I've touched mt7620 devices so I forgot it has the trigger or not... tpt is prefered to radio, of course.

Reply:
No worries! Keep an eye on me ... seriously 😆